src/features/query-bar/components/filter/FilterButton.js

```diff
@@ -51,17 +51,21 @@ class FilterButton extends React.Component<Props, State> {
 
     componentDidUpdate(prevProps: Props, prevState: State) {
         const { columns } = this.props;
-        const { isMenuOpen } = this.state;
+        const { isMenuOpen, transientConditions } = this.state;
         const { isMenuOpen: prevIsMenuOpen } = prevState;
         const wasFlyoutOpened = isMenuOpen && !prevIsMenuOpen;
         if (wasFlyoutOpened) {
             const areConditionsEmpty = this.props.conditions.length === 0;
-            if (areConditionsEmpty) {
-                const transientConditions = columns && columns.length === 0 ? [] : [this.createCondition()];
+            const isInitialConditionSet = transientConditions.length > 0;
```

Comment:
This seems like it could be relevant information needed by the component at a later date, 
instead of defining it in a variable inside a lifecycle method, 
would it make more sense to define it in state?

Reply:
It looks like the only usage in <FilterButton /> where the length is being checked is on L309 so it doesn't make too much sense to set it in state. Could you elaborate more on why this information would be relevant at a later date?

Reply:
There are there cases where this value is being used. 
isInitialConditionSet, shouldSetInitialCondition, shouldReinitializeConidtion, 
Your saying that there will never be another case that these checks won't be used?

Reply:
So after the logic cleanup it's only being used in one spot - L61.